content/docs/work-with-source-git.md

```diff
@@ -0,0 +1,126 @@
+draft: false
+disableToc: false
+weight: 12
+aliases:
+- /source-git
+---
+
+This document is focused on how packit treats a source-git repo.
+
+If you are looking for the design concept of source-git, please head on here:
+[docs/source-git-design]({{< ref "/docs/source-git-design.md" >}})
+
+In short, source-git concept is fairly simple - take an upstream repo and shove
+downstream content on top of it. Easy right?
+
+Packit is then able to work with such a repo and:
+
+* Create SRPMs
+
+* Sync the state into dist-git
+
+* And you are able to utilize all features of packit-service: builds on PRs,
+  tests on those builds, update your package in Fedora.
+
+Let's describe all of these in detail.
+
+
+### Upstream history or not
+
+Your source-git repo can contain upstream git history if you want. Or it
+doesn't. This is really up to you and packit is able to work with both.
+
+The key ingredient here is the config option `upstream_ref`. It tells packit
+the git ref which is suppose to match the archive specified in the spec file.
+All commits on top of this ref are treated either as:
+
+* downstream patches
+* downstream packaging
+
+For downstream patches, packit creates a patch from git commits on top of
+`upstream_ref`. Downstream packaging changes are ignored in patch files since
+they can be directly committed to dist-git. You can control the ignore
+mechanism with config option `patch_generation_ignore_paths`.
+
+
+#### Example
+
+Let's have a look at source-git repo for pacemaker package:
+```
+$ git log --online
+bd12722 (HEAD -> c8s-test, origin/c8s-test, master) downstream packaging
+7201e28 Refactor: controller: remove unused argument
+91c557c Refactor: controller: convert active_op_t booleans to bitmask
+2e90063 Refactor: controller: rename struct recurring_op_s to active_op_t
+3ee90ce (tag: source-git-start) pacemaker-2.0.3 base
+```
+
+We have 5 commits:
+* bd12722 - top commit, contains a spec file so packit won't create a patch file out of this one
+* 7201e28 - a downstream patch
+* 91c557c - a downstream patch
+* 2e90063 - a downstream patch
+* 3ee90ce - the bottom commit, equals to an unpackaed upstream tarball
+
+packit.yaml has only two lines:
+```
+$ cat .packit.yaml
+specfile_path: SPECS/pacemaker.spec
+upstream_ref: source-git-start
+```
+
+As you can see, we are telling packit, that commit tagged as `source-git-start`
+indicates the barrier between upstream and downstream, in our case it's just a
+single commmit with the upstream archive content, but it could have been the
+whole upstream git history.
+
+
+### Adding changes
+
+So, how can one add new changes into a source-git repo? That's simple - just
+commit them. If your repo is on GitHub (or another public forge), you can even
+force-push changes to achieve desired results with patches.
+
+The important point with patch files is, that a commit which changes code will be:
+1. converted into a patch file (using `git format-patch`)
+2. and added to the spec file
```

Comment:
```suggestion
2. added to the spec file
```

Reply:
I'm sorry but I like mine more (even though it may not be so correct)

Reply:
I'm mentioning it here, but probably we should adopt a documentation style guide for our projects. [Google has one](https://developers.google.com/style/tense), but then [here is a collection](https://www.writethedocs.org/guide/writing/style-guides/#selecting-a-good-style-guide-for-you) of them, which reveals, that [Red Hat also has one](https://stylepedia.net/style/).

I've added this as a discussion point for next Tuesday.

This comment is not a blocker for this PR.